ci/publish_github_release.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+# NOTE: please do not set trace mode (-x)
+# as it will leak credentials in the logs.
+set -e
+
+if [[ -z "${CIRCLE_TAG:-}" ]]; then
+    echo "skipping non tag events"
+    exit 0
+fi
+
+if [[ ! -f "${ENVOY_SRCDIR}/build_release_stripped/envoy" ]]; then
+    echo "could not locate envoy binary at path: ${ENVOY_SRCDIR}/build_release_stripped/envoy"
+    exit 1
+fi
+
+if [[ -z "${GITHUB_TOKEN:-}" ]]; then
+    echo "environment variable GITHUB_TOKEN unset; cannot continue with publishing."
+    exit 1
+fi
+
+if [[ -z "${GITHUB_USER:-}" ]]; then
+    echo "environment variable GITHUB_USER unset; cannot continue with publishing."
+    exit 1
+fi
+
+if [[ -z "${GITHUB_REPO:-}" ]]; then
+    echo "environment variable GITHUB_REPO unset; cannot continue with publishing."
+    exit 1
+fi
+
+wget https://github.com/aktau/github-release/releases/download/v0.7.2/linux-amd64-github-release.tar.bz2 -O /tmp/ghrelease.tar.bz2
+tar -xvjpf /tmp/ghrelease.tar.bz2 -C /tmp
+cp /tmp/bin/linux/amd64/github-release /usr/local/bin/ghrelease
+chmod +x /usr/local/bin/ghrelease
+
+ghrelease release --tag "${CIRCLE_TAG:-}" --name "${CIRCLE_TAG:-}"
```

Comment:
@taion809 From a cursory look, this appears to create a Github release?

Currently, I think the process is roughly as follows:

1. @mattklein123 manually creates a release on Github
2. CircleCI receives a webhook with `$CIRCLE_TAG` set to the tag e.g. v1.5.0
3. Scripts that need to do things on releases, like publish Docker images, check for the existence of `$CIRCLE_TAG`.

So I think this needs to be changed to add the binary to the existing Github release, otherwise we'll need to revise the release process for Envoy.